File: docs/examples/server-webhook-setup.ts
/**
 * Register a webhook endpoint to receive fikashop payment and subscription events.
 *
 * Requires the server admin token (FIKASHOP_ADMIN_ACCESS_TOKEN) — never run
 * this from client bundles (React Native, Expo, browser).
 *
 * After registration, fikashop delivers Stripe-style event envelopes to your URL.
 * See webhook-host-handler.ts for a receiver example.
 */
import { createFikashopClient } from '@fikashop/payment-gateway-client';

async function registerWebhookEndpoint() {
  const adminToken = process.env.FIKASHOP_ADMIN_ACCESS_TOKEN;
  const baseUrl = process.env.FIKASHOP_API_URL ?? 'https://api.fikashop.app';
  const partnerCode = process.env.FIKASHOP_PARTNER_CODE ?? '';
  const webhookUrl = process.env.FIKASHOP_WEBHOOK_URL ?? 'https://your-app.example/webhooks/fikashop';
  const webhookSecret = process.env.FIKASHOP_WEBHOOK_SECRET ?? 'whsec_change_me';

  if (!adminToken) {
    throw new Error('Set FIKASHOP_ADMIN_ACCESS_TOKEN (dashboard Settings → API keys)');
  }
  if (!partnerCode) {
    throw new Error('Set FIKASHOP_PARTNER_CODE for X-Partner-Id');
  }

  const client = createFikashopClient({
    baseUrl,
    getAccessToken: async () => adminToken,
    partnerId: partnerCode,
  });

  // Register — empty subscribed_events means all event types
  const resp = await client.post('/shop/api/admin/webhooks/endpoints/', {
    url: webhookUrl,
    secret: webhookSecret,
    enabled: true,
    is_default: true,
    subscribed_events: [],
  });

  if (!resp.ok) {
    throw new Error(resp.problem ?? `Webhook registration failed (${resp.status})`);
  }

  // Response includes id, uuid, url, enabled, is_default, subscribed_events, created_at
  // secret is write-only and never returned
  return resp.data;
}

registerWebhookEndpoint().catch(console.error);

File: docs/examples/server-webhook-setup.py
#!/usr/bin/env python3
"""
Register a webhook endpoint to receive fikashop payment and subscription events.

Requires the server admin token (FIKASHOP_ADMIN_ACCESS_TOKEN) — never run
this from client bundles (React Native, Expo, browser).

After registration, fikashop delivers Stripe-style event envelopes to your URL.
See fastapi_webhook.py or django_reference.md for receiver examples.
"""

from __future__ import annotations

import os
import sys

import requests

BASE_URL = os.environ.get("FIKASHOP_API_URL", "https://api.fikashop.app").rstrip("/")
ADMIN_TOKEN = os.environ.get("FIKASHOP_ADMIN_ACCESS_TOKEN", "")
PARTNER_CODE = os.environ.get("FIKASHOP_PARTNER_CODE", "")
WEBHOOK_URL = os.environ.get("FIKASHOP_WEBHOOK_URL", "https://your-app.example/webhooks/fikashop")
WEBHOOK_SECRET = os.environ.get("FIKASHOP_WEBHOOK_SECRET", "whsec_change_me")

ENDPOINTS_URL = f"{BASE_URL}/shop/api/admin/webhooks/endpoints/"


def headers() -> dict[str, str]:
    if not ADMIN_TOKEN:
        raise SystemExit("Set FIKASHOP_ADMIN_ACCESS_TOKEN (dashboard Settings → API keys)")
    if not PARTNER_CODE:
        raise SystemExit("Set FIKASHOP_PARTNER_CODE")
    return {
        "Authorization": f"Bearer {ADMIN_TOKEN}",
        "Content-Type": "application/json",
        "X-Partner-Id": PARTNER_CODE,
    }


def register_webhook_endpoint() -> dict:
    """Register a webhook endpoint. Returns the created endpoint (no secret in response)."""
    payload = {
        "url": WEBHOOK_URL,
        "secret": WEBHOOK_SECRET,
        "enabled": True,
        "is_default": True,
        "subscribed_events": [],  # empty = all event types
    }
    resp = requests.post(ENDPOINTS_URL, json=payload, headers=headers(), timeout=30)
    resp.raise_for_status()
    return resp.json()


def main() -> None:
    endpoint = register_webhook_endpoint()
    print(f"Registered webhook endpoint {endpoint['uuid']}")
    print(f"  URL: {endpoint['url']}")
    print(f"  Enabled: {endpoint['enabled']}")
    print(f"  Subscribed events: {endpoint['subscribed_events'] or '(all)'}")


if __name__ == "__main__":
    try:
        main()
    except requests.HTTPError as exc:
        print(exc.response.text if exc.response is not None else exc, file=sys.stderr)
        raise
